pkg/volume/gce_pd/gce_util.go

```diff
@@ -149,11 +150,59 @@ func (mounter *gceSafeFormatAndMount) Mount(source string, target string, fstype
 	}
 	args = append(args, options...)
 	args = append(args, source, target)
-	glog.V(5).Infof("exec-ing: /usr/share/google/safe_format_and_mount %v", args)
-	cmd := mounter.runner.Command("/usr/share/google/safe_format_and_mount", args...)
+	safeFormatAndMountPath := "/usr/share/google/safe_format_and_mount"
+	glog.V(5).Infof("exec-ing: %s %v", safeFormatAndMountPath, args)
+	cmd := mounter.runner.Command(safeFormatAndMountPath, args...)
 	dataOut, err := cmd.CombinedOutput()
 	if err != nil {
-		glog.V(5).Infof("error running /usr/share/google/safe_format_and_mount\n%s", string(dataOut))
+		glog.V(5).Infof("error running %s\n%s", safeFormatAndMountPath, string(dataOut))
+		// if safe_format_and_mount is not available on this system use linux utilities directly.
+		if strings.Contains(err.Error(), safeFormatAndMountPath+": no such file or directory") {
+			return mounter.formatAndMount(source, target, fstype, options)
+		}
+	}
+	return err
+}
+
+// Uses unix utils to format and mount the given disk
+func (mounter *gceSafeFormatAndMount) formatAndMount(source string, target string, fstype string, options []string) error {
+
+	// Add the default options from safe_format_and_mount
+	options = append(options, "discard")
+	options = append(options, "defaults")
+
+	// Try to mount the disk. Let 'mount' try to determine the fstype.
+	err := mounter.Interface.Mount(source, target, "", options)
+	if err != nil && strings.Contains(err.Error(), "unknown filesystem type '(null)'") {
```

Comment:
It is the combined failure of 'mount' with 'filesystem type (null)' and 'file' returning 'data' for the file that leads to the conclusion that disk is unformatted. If we were to call diskLooksUnformatted first we would still have to call mount to corroborate as 'file' can return 'data' for a file it does not recognized even if it is a formatted device. I don't think that is likely but that is what the man page states. the safe_format_and_mount script works the same way.
Also, in the life cycle of the disk it will only be formatted once but possibly mounted many times making the first mount call more likely to succeed and the 'diskLooksUnformatted' call unnecessary.